dattri/func/projection.py

```diff
@@ -333,50 +404,27 @@ def project(
                 of batch of features.
             ensemble_id (int): A unique ID for this ensemble.
 
-        Raises:
-            RuntimeError: The batch size of the CudaProjector is too large for
-                your GPU.
-            RuntimeError: Too many resources requested for launch CUDA.
-
         Returns:
             Tensor: The projected features.
         """
-        import fast_jl
+        # Regenerate randomness if the ensemble_id has changed
+        if ensemble_id != self.current_ensemble_id:
+            self._generate_randomness(ensemble_id)
+            self.current_ensemble_id = ensemble_id
 
         if isinstance(features, dict):
             features = vectorize(features, device=self.device)
-        elif features.device.type != self.device:
+        if features.device != self.device:
             features = features.to(self.device)
-        batch_size = features.shape[0]
-
-        effective_batch_size = 32
-        min_proj_batch_size = 8
-        if batch_size <= min_proj_batch_size:
-            effective_batch_size = min_proj_batch_size
-        elif batch_size <= min_proj_batch_size * 2:
-            effective_batch_size = min_proj_batch_size * 2
+        if features.dtype != self.dtype:
+            features = features.to(self.dtype)
 
-        effective_batch_size = min(self.max_batch_size, effective_batch_size)
+        if self.proj_type == ProjectionType.sjlt:
+            with torch.no_grad():
+                result = self.sjlt(features)
 
-        function_name = f"project_{self.proj_type}_{effective_batch_size}"
-
-        fn = getattr(fast_jl, function_name)
-
-        try:
-            result = fn(
-                features,
-                self.proj_dim,
-                self.seed + int(1e4) * ensemble_id,
-                self.num_sms,
-            )
-        except RuntimeError as e:
-            if "CUDA error: too many resources requested for launch" in str(e):
-                # provide a more helpful error message
-                msg = "The batch size of the CudaProjector is too large for your GPU. \
-                    Reduce it by using the proj_max_batch_size argument.\
-                    \nOriginal error:"
-                raise RuntimeError(msg) from e
-            raise e from None
+        elif self.proj_type in [ProjectionType.rademacher, ProjectionType.normal]:
+            result = features @ self.proj_matrix / (self.proj_dim**0.5)
```

Comment:
Correct me if I'm wrong. If we use cudaprojector with non-sjlt types (like rademacher and normal). Th e implementation falls back to a naive projection (w/o block size in basic projector)